### DISCUSSION TITLE
state space model of the system

### DISCUSSION THREAD

**Discussion by frostyduck** [NREL-SIIP/PowerSimulationsDynamics.jl#297] [Q&A]
How can I construct a state space model of the system (matrix A,B,C,D) using PowerSimulationsDynamics.jl? Just I want to change a branch impedance in one line with of a power system with step 0.001 from 0 to 0.95 (using BranchImpedanceChange) in order to get sequence of ABCD matrices.

**Comment by jd-lara** [comment]
@frostyduck PowerSimulationsDynamics isn't built to return A, B, C, D matrices easily. This would be a feature request. 

For your use case, you can use the perturbation described here https://nrel-siip.github.io/PowerSimulationsDynamics.jl/stable/api/public/#PowerSimulationsDynamics.BranchImpedanceChange 

PowerSimulationsDynamics.jl is built to handle perturbations via callbacks.

**Reply by frostyduck** [reply]
@jd-lara thank you for your prompt response.

> PowerSimulationsDynamics isn't built to return A, B, C, D matrices easily. This would be a feature request.

But it can be to possible? "not easily"? Maybe through some interactions Matlab toolboxes (pst/psat)?

**Reply by jd-lara** [reply]
Well the design of PSID isn't really set-up for that kind of interaction with other softwares. But we can try to enable the generation of the matrices are requested.
